src/Avalonia.Base/Rendering/Composition/Server/ServerCompositionTarget.cs

```diff
@@ -163,7 +163,7 @@ public void Render()
                     _dirtyRect = new Rect(0, 0, layerSize.Width, layerSize.Height);
                 }
 
-                if (!_dirtyRect.IsDefault)
+                if (_dirtyRect.Width != 0 || _dirtyRect.Height != 0)
```

Comment:
Please read this: https://github.com/AvaloniaUI/Avalonia/issues/10264#issuecomment-1471014426. `Rect` has special considerations.